### PR TITLE
add page404 and validate URL

### PR DESCRIPTION
**Pull request by zhouzilong2020**
截屏2022-04-29 16 08 58

### CODE REVIEW

File: frontend/src/views/Page404.vue [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<template>
```
Comment (tianzhou) [inline frontend/src/views/Page404.vue:1]: Do we still need this?

File: frontend/src/router/index.ts [hunk 2]
```diff
@@ -12,11 +23,38 @@ export type RouteParam = {
   minRAM: number;
   keyword: string;
 };
+
+export const RouteParamDefault: RouteParam = {
+  cloudProvider: SearchConfigDefault.cloudProvider,
```
Comment (tianzhou) [inline frontend/src/router/index.ts:28]: please update the query parameter per offline discussion.

File: frontend/src/views/Page404.vue (hunk 1 above)
Comment (zhouzilong2020) [reply frontend/src/views/Page404.vue:1]: Just in case user type in some wrong URL. I incline to keep the 404 page.

**Comment by zhouzilong2020** [comment]
PTAL

**Review by LiuJi-Jim — COMMENTED** [review]
LGTM

**Review by tianzhou — APPROVED** [review]
LGTM

**Pull request merged**